contracts/EVMCode.slb

```diff
@@ -6,8 +6,48 @@ library EVMCode {
 
     struct Code {
         address codeAddress;
-        bytes codeBytes;
         uint length;
+        DataNode root;
+    }
+
+    struct DataNode {
+        uint pos;
+        uint value;
+        uint next;
+    }
+
+    struct RawCode {
+        uint pos;
+        uint value;
+    }
+
+    function nextDataNode(DataNode memory node) internal pure returns (DataNode memory next) {
+        if (node.next == 0) {
+            return DataNode(uint(-1), 0, 0);
+        }
+
+        assembly {
+            next := mload(add(node, 0x40))
+        }
+    }
+
+    function fromArray(RawCode[] memory codes, uint length) internal pure returns (Code memory code) {
+        require(codes.length >= 1, "Empty code array");
+
+        code.length = length;
+        code.root = DataNode(codes[0].pos, codes[0].value, 0);
+
+        DataNode memory current = code.root;
+        for (uint i = 1; i < codes.length; i++) {
```

Comment:
We need more sanity checks here, otherwise some parts a exploitable (if not checked elsewhere)